pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -348,6 +348,11 @@ func (plugin *kubenetNetworkPlugin) setup(namespace string, name string, id kube
 	if err != nil {
 		return err
 	}
+
+	// This also add the current pod to the running pods list because for rkt, when we set up the network,
+	// the pod is not currently running.
+	runningPods = append(runningPods, &hostport.RunningPod{Pod: pod, IP: ip4})
```

Comment:
@freehan The docker case is fine as later the host ports are gathered in a map. https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/network/hostport/hostport.go#L134